runtime/src/append_vec.rs

```diff
@@ -33,7 +33,10 @@ macro_rules! u64_align {
     };
 }
 
+#[cfg(target_pointer_width = "64")]
 const MAXIMUM_APPEND_VEC_FILE_SIZE: usize = 16 * 1024 * 1024 * 1024; // 16 GiB
```

Comment:
Do you know how much blows up if we just retype `MAXIMUM_APPEND_VEC_FILE_SIZE` as a `u64`?   It's basically a certainty that we'll never fully support 32-byte targets for the runtime, so I think I'd rather take that approach then maintain a separate `MAXIMUM_APPEND_VEC_FILE_SIZE` value for 32-byte targets that doesn't really do anything practical

Reply:
From what I can tell, it's used for an `mmap` (so essentially cast to a `Vec`). There's no way to index past 4GB on 32-bit architectures, so the indexing might overflow if it's retyped as a `u64`.

It _could_ work if we choose to bail some other way on 32-bit platforms, e.g. panic when casting would otherwise overflow. Again, I'm not too familiar with this code so I might not be the best one to be making this decision :sweat_smile:

Reply:
That mmap code should never run on a 32bit platform as we don’t support 32bit in the runtime at all, so if a `u64` allows your build to compile and doesn’t cause too many other nasty typecasts elsewhere in runtime/ then I think that’s the way to go

Reply:
If that's the case, I'd feel more comfortable making the relevant modules conditionally compiled. I gave a quick look at doing that, although it looks like there's a non-trivial number of imports pulled in from `runtime` in `transaction-status` (which is where `client` pulls it in through).

At any rate, I just pushed a change to make it a `u64` instead, and it'll return the error case if it happens not to fit in `usize`.